javascript/javascript3/week1/homework/giphy-api/giphy-api.js

```diff
@@ -0,0 +1,42 @@
+
+// Giphy app 
+const apiKey = "6oiwHyH3t5u02qqbm5cTIk0BKpE2zGV8";
+const defaultGifLimit = 10;
+
+const searchingForGifElement = document.getElementById("searchingForGif");
+const searchButtonElement = document.getElementById("searchButton");
+const numbersOfGifElement = document.getElementById("numbersOfGif");
+const gifListsContainerElement = document.querySelector(".gifListsContainer")
+
+searchButtonElement.addEventListener("click", () => {
+    searchingForGifElement.innerHTML = ""
+    const searchedWord = searchingForGifElement.value.toLowerCase().trim();
+    if(searchedWord == ""){
+        searchingForGifElement.innerHTML = "Please enter a word!"
+    }
+    let gifLimit = parseInt(numbersOfGifElement.value.trim());
+    if(isNaN(gifLimit)){
+        gifLimit = defaultGifLimit;
+    };
+    fetchGiphy(searchedWord, gifLimit);
+})
+
+function fetchGiphy(searchedWord, gifLimit){
+    fetch(`https://api.giphy.com/v1/gifs/search?api_key=${apiKey}&q=${searchedWord}&limit=${gifLimit}&offset=0&rating=g&lang=en`)
```

Comment:
When this fails, the user will get an error that is not very descriptive.
It is good add an error handler that either retries a couple of times, or informs the user that something is wrong.

Reply:
The promise will also not fail on http 404 or 500 response, but the 'ok' property will just be false. Check for this and show something suitable :)